interpreter/luajit/trace.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2024 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+package luajit
+
+import (
+	"encoding/binary"
+	"errors"
+	"hash/fnv"
+
+	"github.com/open-telemetry/opentelemetry-ebpf-profiler/libpf"
+	"github.com/open-telemetry/opentelemetry-ebpf-profiler/remotememory"
+)
+
+// This offset is the same in arm64/x86_64 for all known versions of luajit.
+// (gdb) p &((GCtrace*)0)->startins
+// $7 = (uint16_t *) 0x50
+const tracePartOffset = 0x50
+
+type jitStatePart struct {
+	trace     libpf.Address
+	_         uint32 // freetrace
+	sizetrace uint32
+}
+
+type trace struct {
+	_       uint32 /* startins Original bytecode of starting instruction. */
+	szmcode uint32 /* Size of machine code. */
+	mcode   uint64 /* Start of machine code. */
+	_       uint32 /* mcloop */
+	// TODONE: For arm is LJ_ABI_PAUTH defined?
+	// For docker images at least LJ_ABI_PAUTH is not defined.
+	// ASMFunction mcauth;	/* Start of machine code, with ptr auth applied. */
+	_        uint16 /* Number of child traces (root trace only). */
+	spadjust uint16 /* Stack pointer adjustment (offset in bytes). */
+	traceno  uint16 /* Trace number. */
+	_        uint16 /* Linked trace (or self for loops). */
+	root     uint16 /* Root trace of side trace (or 0 for root traces). */
+}
+
+type traceMap map[uint16]trace
+
+func hashTraces(tracesAddr libpf.Address, rm remotememory.RemoteMemory) (
```

Comment:
Maybe name this something like `getAndHashTraceAddrs`, since it's not just doing a pure hashing computation, it's also reading the trace addrs from program memory.

Reply:
:+1: